var/spack/repos/builtin/packages/py-python-ldap/package.py

```diff
@@ -23,4 +25,8 @@ class PyPythonLdap(PythonPackage):
     depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
     depends_on('py-pyasn1@0.3.7:', type=('build', 'run'))
     depends_on('py-pyasn1-modules@0.1.5:', type=('build', 'run'))
-    depends_on('cyrus-sasl', type='link')
+    depends_on('cyrus-sasl', type='link', when='^openldap+sasl')
+
+    def patch(self):
+        if self.spec.satisfies('^openldap~sasl'):
+            filter_file('HAVE_SASL ', '', 'setup.cfg')
```

Comment:
(question) Do you know if an option is exposed to `setup.py`? E.g.

```
var/spack/repos/builtin/packages//py-h5py/package.py:        self.setup_py('configure', '--hdf5={0}'.format(spec['hdf5'].prefix),
var/spack/repos/builtin/packages//py-h5py/package.py-                      '--hdf5-version={0}'.format(spec['hdf5'].version))
```

Or @adamjstewart do you know if there is a preferred way to set something like this?

Reply:
The preferred way is to not pass options to `setup.py`. Once #27798 is merged, we won't even be running `setup.py` anymore. Environment variables or patches are fine.

Reply:
@scheibelp it is not exposed